Python script for playing sounds on RPi for E2E tests

KritsanaMike opened the pull request:
python script for playsound

KritsanaMike committed:
python script for playsound

antonyharfield commented on playsound/play_sound.py:
```diff
@@ -0,0 +1,26 @@
+        if t_min == 00 or t_min == 30:
+            print(t_hr)
+            play_sound()
+        time.sleep(1)
```
Should sleep be called on every iteration of the while loop?

antonyharfield commented on playsound/play_sound.py:
```diff
@@ -0,0 +1,26 @@
+import random
+
+def play_sound():
+    name = ['dog','truck','motorcycle','chainsaw']
```
If you hardcode the files then you need to change the script for every new file. Instead, scan the current directory for wav files.

antonyharfield replied:
You could scan for files when the program starts, and display an error if none are found.

KritsanaMike replied:
I fixed them. check them for me, please.

antonyharfield reviewed (changes requested):
Can we make this part of the CLI? E.g. `cli sounds`

KritsanaMike committed:
fix code playsound

- read filename in folder then random one name to play sound
- move location sleep and change from 1 sec to 10 sec

Tooseriuz commented on playsound/play_sound.py:
```diff
@@ -0,0 +1,35 @@
+
+def readfile():
+    file = os.listdir(path)
+    return file
```
naming `file` as `files` would be better since we expect it to be list of files

Tooseriuz replied:
and function name would be `getFiles` or `readFiles`

KritsanaMike replied:
> naming `file` as `files` would be better since we expect it to be list of files
Ok. I will fix this name

> and function name would be `getFiles` or `readFiles`
and this one too

Tooseriuz commented on playsound/play_sound.py:
```diff
@@ -0,0 +1,35 @@
+    if len(name) > 0:
+        ran = random.choice(name)
+        print("Playing..."+ran)
+        playsound(path+ran)
```
- since we are getting list of files then `name` should be `files`
- `ran` should be `file` since we randomly pick a file
- better to have white space between `+` `print("Playing..." + file)`

Tooseriuz commented on playsound/play_sound.py:
```diff
@@ -0,0 +1,35 @@
+while True:
+    result = time.localtime()
+    t_hr = result.tm_hour
+    t_min = result.tm_min
```
- better to have white space between + 
- naming `result` to `system_time` or `local_time`

Tooseriuz commented on playsound/play_sound.py:
```diff
@@ -0,0 +1,35 @@
+        if t_min == 0:
+            print(t_hr)
+            play_sound()         
+    time.sleep(10)
```
you could refactor logic to 
```
if t_hr > 22:
   continue
if t_min != 0 or t_min != 30:
   continue
if t_hr == 22 and t_min > 0:
   continue
print(t_hr)
play_sound() 
```

KritsanaMike replied:
> you could refactor logic to
> 
> ```
> if t_hr > 22:
>    continue
> if t_min != 0 or t_min != 30:
>    continue
> if t_hr == 22 and t_min > 0:
>    continue
> print(t_hr)
> play_sound() 
> ```

I think this logic does not support, cause we have to start play at 9 AM and finish at 10 PM. but your logic that look like play every time is not more than 10 PM

Tooseriuz reviewed (changes requested):
some naming and refactor for better readability

KritsanaMike committed:
fix code from reviewer

- fix Function name
- fix variable name

KritsanaMike committed:
fix if condition
